tests/flow/test_call_subquery.py

```diff
@@ -42,30 +42,104 @@ def expect_error(self, query, expected_err_msg):
+        for query in queries:
+            self.expect_error(query, expected_error)
+
+        # invalid queries: MATCH without preceding WITH
+        queries = [
+            "CALL {CREATE (n:N) MATCH (n:N) RETURN n} RETURN 1",
+            "WITH 1 AS a CALL {WITH a CREATE (n:N) MATCH (n:N) RETURN n} RETURN a",
+        ]
+        expected_error = "A WITH clause is required to introduce MATCH after an updating clause"
+        for query in queries:
+            self.expect_error(query, expected_error)
+
+        # invalid queries: redeclaring variable in CREATE
+        queries = [
+            "CALL {MATCH (n:N) CREATE (n:N2)} RETURN 1 ",
+        ]
+        expected_error = "The bound variable 'n' can't be redeclared in a CREATE clause"
+        for query in queries:
+            self.expect_error(query, expected_error)
+
+        # invalid queries: UNION column names
+        queries = [
+            # different column names
+            """
+            CALL {
+                RETURN 1 AS one
+                UNION
+                RETURN 2 AS two
+            }
+            RETURN 1
+            """,
+            """
+            MATCH (n)
+            CALL {
+                RETURN 1 AS one
+                UNION
+                RETURN n
+            } RETURN 1
+            """,
+            # missing column names
+            """
+            CALL {
+                RETURN 1
+                UNION
+                RETURN 2 AS two
+            } RETURN 1
+            """,
+        ]
+        expected_error = "All sub queries in a UNION must have the same column names"
+        for query in queries:
+            self.expect_error(query, expected_error)
+
+        # invalid queries: wrong number of UNION clauses
+        queries = [
+            "CALL {WITH 1 UNION RETURN 1} RETURN 0"
+        ]
+        expected_error = "Found 1 UNION clauses but only 1 RETURN clauses"
+        for query in queries:
+            self.expect_error(query, expected_error)
+
+        # invalid queries: subquery with invalid termination
+        queries = [
+            "CALL {MATCH (n)} RETURN 0",
+            "CALL {RETURN 1 UNION WITH 1} RETURN 0"
         ]
-        for query, err in queries_errors:
-            self.expect_error(query, err)
+        expected_error = "Query cannot conclude with"
+        for query in queries:
+            self.expect_error(query, expected_error)
+
+        # invalid queries: returning literals inside CALL{} must be aliased
+        queries = [
+            "CALL {RETURN 1} RETURN 0",
+            "CALL {RETURN 'a'} RETURN 0",
+            "CALL {RETURN 5.7} RETURN 0",
+            "CALL {RETURN 1+2} RETURN 0",
+            "CALL {RETURN true} RETURN 0",
+            "CALL {RETURN sum(0)} RETURN 0",
+            "CALL {MATCH (n) RETURN n.x} RETURN 0",
+            "CALL {MATCH p=(n) RETURN nodes(p)[0]} RETURN 0",
+        ]
+        for query in queries:
+            self.expect_error(query, "must be aliased")
```

Comment:
I'll assume the error message gives a bit more information than the string `must be aliased`.

Reply:
A correct assumption :)